File: java/hole7/src/test/java/hole7/TakeHomeCalculatorShould.java
package hole7;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static hole7.Money.money;
import static hole7.TaxRate.taxRate;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

@DisplayName("Hole 7: Take home calculator should")
public class TakeHomeCalculatorShould {

    @Test
    public void calculate_tax() {
        Integer first = new TakeHomeCalculator(taxRate(10)).netAmount(
                money(40, "GBP"),
                money(50, "GBP"),
                money(60, "GBP")
        ).value;
        assertEquals(135, first.intValue());
    }

    @Test
    public void not_sum_different_currencies() {
        assertThrows(Incalculable.class, () ->
                new TakeHomeCalculator(taxRate(10)).netAmount(
                        money(4, "GBP"),
                        money(5, "USD"))
        );
    }
}

File: kotlin/hole7/src/test/kotlin/TakeHomeCalculatorShould.kt
package hole7

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

@DisplayName("Hole 7: Take home calculator should")
class TakeHomeCalculatorShould {
    @Test
    fun `calculate tax`() {
        val first = TakeHomeCalculator(TaxRate(10)).netAmount(
                Money(40, "GBP"),
                Money(50, "GBP"),
                Money(60, "GBP")
        ).value
        assertEquals(135, first)
    }

    @Test
    fun `not sum different currencies`() {
        assertThrows<Incalculable> {
            TakeHomeCalculator(TaxRate(10)).netAmount(
                    Money(4, "GBP"),
                    Money(5, "USD")
            )
        }
    }
}
